lib/elastic_apm/error_builder.rb

```diff
@@ -40,7 +37,7 @@ def build_log(message, context: nil, backtrace: nil, **attrs)
 
     def add_stacktrace(error, kind, backtrace)
       stacktrace =
-        @agent.stacktrace_builder.build(backtrace, type: :error)
+          ElasticAPM.agent.stacktrace_builder.build(backtrace, type: :error)
```

Comment:
Just a thought: Should we do a brief check for the agent not being `nil` at the point of initialising any class that depends on it being present, like this one?

Like,

```ruby
def initialize(...)
  raise AgentRequired unless ElasticAPM.agent
  
  # ...
end
```

As a safety measure for ourselves and, hopefully, catching cases where the agent is missing as early as possible.

Reply:
Yes, I'm thinking I need to be more defensive if there is no agent when the config is accessed in general. For example, having a method on `ElasticAPM` that takes a config option name so that such an error can be raised if there is no agent and thus no config.

```ruby
module ElasticAPM
    def get_config(option)
      raise AgentRequired unless agent
      Agent.config.send(option)
    end
```

instead of just calling `ElasticAPM.config.option_name` everywhere. I would be afraid we'd get a `no method on NilClass` error at some point.

Reply:
What do you think of that?

Reply:
```ruby
module ElasticAPM
    def config
      raise AgentRequired unless agent
      Agent.config
    end
end
```

Would essentially be the same but allow us to use ordinary method chaining like `ElasticAPM.config.option` without the `send`.

Reply:
Yeah, but what is the config is `nil` for some reason?

Reply:
The we could also raise.

```
module ElasticAPM
  def config
    raise AgentRequired unless agent
    raise RunTimeError, 'mysteriously missing Config' unless agent.config
    Agent.config
  end
end
```